specs/BrowserProcessExited.md

```diff
@@ -0,0 +1,186 @@
+        [browserExitedEventToken, this](
+            ICoreWebView2Environment* sender,
+            ICoreWebView2BrowserProcessExitedEventArgs* args) {
+        COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND kind;
+        CHECK_FAILURE(args->get_BrowserProcessExitKind(&kind));
+
+        // Watch for graceful browser process exit. Let ProcessFailed event
+        // handler take care of failed browser process termination.
+        if (kind == COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND_NORMAL_EXIT)
+        {
+          CHECK_FAILURE(
+              m_webViewEnvironment->remove_BrowserProcessExited(browserExitedEventToken));
+          // Release the environment only after the handler is invoked.
+          // Otherwise, there will be no environment to raise the event when
+          // the process exits.
+          m_webViewEnvironment = nullptr;
+          CleanupUserDataFolder();
+        }
+
+        return S_OK;
+    }).Get(),
+    &browserExitedEventToken));
+```
+
+## .NET C#
+```c#
+// Get the environment from the CoreWebView2 and add a handler.
+webView.CoreWebView2.Environment.BrowserProcessExited += Environment_BrowserProcessExited;
+
+// Check and report browser process exit kind.
+void Environment_BrowserProcessExited(object sender, CoreWebView2BrowserProcessExitedEventArgs e)
+{
+    var exitKind = (e.BrowserProcessExitKind == CoreWebView2BrowserProcessExitKind.NormalExit) ? "normally" : "unexpectedly";
+    MessageBox.Show(this, $"The browser process has exited {exitKind}.", "Browser Process Exited");
+}
+```
+
+
+# Remarks
+Note this is an event from the `ICoreWebView2Environment` interface, not the
+`ICoreWebView2`. The difference between this `BrowserProcessExited` event and
+the `CoreWebView2`'s `ProcessFailed` event is that `BrowserProcessExited` is
+raised for any (expected and unexpected) **browser process** exits, while
+`ProcessFailed` is raised only for **unexpected** browser process exits, or for
+**render process** exits/unresponsiveness. To learn more about the WebView2
+Process Model, go to (Process model)[https://docs.microsoft.com/en-us/microsoft-edge/webview2/concepts/process-model].
+
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+
+# API Details
+
+## COM
+```cpp
+library WebView2
+{
+// ...
+
+/// Specifies the browser process exit type used in the
+/// `ICoreWebView2StagingBrowserProcessExitedEventArgs` interface.
+typedef enum COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND {
+  /// Indicates that the browser process ended normally.
+  COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND_NORMAL_EXIT,
+
+  /// Indicates that the browser process ended unexpectedly.
+  /// A `ProcessFailed` event will also be sent to listening WebViews from the
+  /// `ICoreWebView2Environment` associated to the failed process.
+  COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND_FAILED_EXIT
+} COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND;
+
+interface ICoreWebView2Environment : IUnknown
```

Comment:
This should be like ICoreWebView2Environment2 (or 3?) : ICoreWebView2Environment

Reply:
`ICoreWebView2Environment3 : ICoreWebView2Environment2`.